pkg/cmdlets/auth/login.go

```diff
@@ -178,3 +182,27 @@ func promptToken(credentials *mono_models.Credentials, out output.Outputer, prom
 
 	return nil
 }
+
+// AuthenticateWithDevice attempts to authenticate this device with the Platform.
+func AuthenticateWithDevice(out output.Outputer) error {
+	deviceCode, err := model.RequestDeviceAuthorization()
+	if err != nil {
+		return err
+	}
+	out.Notice(locale.Tr("auth_device_verify_security_code", *deviceCode.UserCode))
+	err = OpenURI(*deviceCode.VerificationURIComplete)
```

Comment:
You should only ever do this in Go if you first nil check the reference. Otherwise you can get a panic if `deviceCode.VerificationURIComplete=nil`.

Reply:
I was looking at the swagger-generated code and it would never do this. A check seemed redundant, but I'll do it anyway. It just likes to pass string pointers in order to reduce allocations.